src/LanguageServer/Protocol/Handler/Extensions/ExtensionMessageResponse.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json.Serialization;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Extensions;
+
+/// <summary>
+/// Return type for the roslyn/extensionWorkspaceMessage and roslyn/extensionDocumentMessage request.
+/// </summary>
+/// <param name="Response">Json response returned by the extension message handler.</param>
+internal readonly record struct ExtensionMessageResponse(
```

Comment:
not certain why we need this type.  can we not just return a string?

Reply:
I think we should use params objects
1.  Sometimes jsonrpc can get confusing if the string passed in is a string to deserialize, or a string parameter.
2.  Params objects mean we can add new fields without it being a breaking change.

Reply:
hrmm..   ok